src/NewsService.Business/CreateNewsCommand.cs

```diff
@@ -17,32 +25,79 @@ public class CreateNewsCommand : ICreateNewsCommand
         private readonly IDbNewsMapper _mapper;
         private readonly INewsValidator _validator;
         private readonly IAccessValidator _accessValidator;
+        private readonly IRequestClient<ICheckDepartmentsExistence> _rcCheckDepartmentsExistence;
+        private readonly ILogger<CreateNewsCommand> _logger;
+
+        private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
+        {
+            if(!departmentId.HasValue)
+            {
+                return null;
+            }
+            string errorMessage = "Failed to check the existing department.";
+            string logMessage = "Department with id {id} not found.";
+
+            try
+            {
+                var response = _rcCheckDepartmentsExistence.GetResponse<IOperationResult<ICheckDepartmentsExistence>>(
+                    ICheckDepartmentsExistence.CreateObj(new List<Guid> { departmentId.Value })).Result;
+                if (response.Message.IsSuccess)
+                {
+                    if (!response.Message.Body.DepartmentIds.Any())
+                    {
+                        errors.Add($"Department Id: {departmentId} does not exist");
+                        return null;
+                    }
+                    return response.Message.Body.DepartmentIds;
+                }
+
+                _logger.LogWarning("Can not find {departmentId} with this Id: {departmentId}: " +
```

Comment:
Can not find department with Id: {departmentId}:

Reply:
fixed